### PR TITLE
[API Compatibility No.154] add target alias for binary_cross_entropy -part

### PR DESCRIPTION
**Pull request by Aidenwu0209**
### PR Category

User Experience

### PR Types

New features

### Description

Support PyTorch-style `target` keyword argument alias for `paddle.nn.functional.binary_cross_entropy` via Python decorator compatibility.

Changes in this PR:
- `python/paddle/nn/functional/loss.py`
  - Add `@param_one_alias(["label", "target"])` on `binary_cross_entropy`
  - Update docstring alias for `label` (`Alias: ``target``.`)
- `test/legacy_test/test_bce_loss.py`
  - Add `test_BCELoss_target_alias` to verify:
    - dygraph `reduction="none"`: `label` and `target` are equivalent
    - dygraph `reduction="mean"`: `label` and `target` are equivalent
    - passing both `label` and `target` raises error

Related Issue: https://github.com/PaddlePaddle/Paddle/issues/76301 (No.154)

### Paddle
- [x] No.154 `torch.nn.functional.binary_cross_entropy -> paddle.nn.functional.binary_cross_entropy`

### PaConvert
- [ ] No.154

### X2Paddle
- [ ] No.154

### d2l
- [ ] No.154

### 是否引起精度变化
否

### CODE REVIEW

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -653,6 +653,9 @@ def binary_cross_entropy(
         label (Tensor): The target labels tensor. 2-D tensor with the same shape as
             ``input``. The target labels which values should be numbers between 0 and 1.
             Available dtype is float16, float32, float64.
+
+            Alias Support: The parameter name ``label`` can be used as an alias for ``target``.
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/loss.py]: 别名文档写简单点就行

Comment (Aidenwu0209) [reply python/paddle/nn/functional/loss.py]: 已按建议精简为一行别名说明：`Alias Support: ``target`` is an alias for ``label``.`，其余逻辑保持不变。

File: python/paddle/nn/functional/loss.py [hunk 2]
```diff
@@ -653,6 +653,8 @@ def binary_cross_entropy(
         label (Tensor): The target labels tensor. 2-D tensor with the same shape as
             ``input``. The target labels which values should be numbers between 0 and 1.
             Available dtype is float16, float32, float64.
+
+            Alias Support: ``target`` is an alias for ``label``.
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/loss.py]: Image

这样

File: test/legacy_test/test_bce_loss.py [hunk 3]
```diff
@@ -238,6 +238,37 @@ def test_BCELoss_error(self):
         )
         paddle.enable_static()
 
+    def test_BCELoss_target_alias(self):
+        paddle.disable_static()
+        try:
```
Comment (zhwesky2010) [inline test/legacy_test/test_bce_loss.py]: 测试代码不要使用try（除非是专门测异常抛出的），需要保证测试case正确可运行

File: python/paddle/nn/functional/loss.py (hunk 2 above)
Comment (Aidenwu0209) [reply python/paddle/nn/functional/loss.py]: 已按截图风格修改为参数别名简写：`Alias: ``target``.`

File: test/legacy_test/test_bce_loss.py [hunk 4]
```diff
@@ -238,6 +238,39 @@ def test_BCELoss_error(self):
...
```
Comment (zhwesky2010) [inline test/legacy_test/test_bce_loss.py]: 测试case不要用try

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

**Comment by luotao1** [comment]
@Aidenwu0209 可以merge下develop，看看能否通过 CI-H 流水线

**Review by zhwesky2010 — APPROVED** [review]
LGTM
